packages/alchemy/src/config/store/client.ts

```diff
@@ -0,0 +1,142 @@
+import { subscribeWithSelector } from "zustand/middleware";
+import { createStore } from "zustand/vanilla";
+import type { AlchemySignerClient } from "../../signer/index.js";
+import {
+  AlchemySigner,
+  type AlchemySignerParams,
+} from "../../signer/signer.js";
+import { AlchemySignerStatus } from "../../signer/types.js";
+import { DEFAULT_IFRAME_CONTAINER_ID } from "../createConfig.js";
+import type { SupportedAccountTypes } from "../types.js";
+import type {
+  AccountState,
+  ClientState,
+  ClientStore,
+  SignerStatus,
+} from "./types.js";
+
+export type CreateClientStoreParams = {
+  client: PartialBy<
+    Exclude<AlchemySignerParams["client"], AlchemySignerClient>,
+    "iframeConfig"
+  >;
+  sessionConfig?: AlchemySignerParams["sessionConfig"];
+};
+
+export const createClientStore = (config: CreateClientStoreParams) => {
+  const clientStore =
+    typeof window === "undefined"
+      ? undefined
+      : createStore(
+          subscribeWithSelector(() => createInitialClientState(config))
+        );
+
+  addClientSideStoreListeners(clientStore);
+
+  return clientStore;
+};
+
+const createSigner = (params: CreateClientStoreParams) => {
+  const { client, sessionConfig } = params;
+  const { iframeContainerId } = client.iframeConfig ?? {
+    iframeContainerId: DEFAULT_IFRAME_CONTAINER_ID,
+  };
+
+  let iframeContainer = document.getElementById(iframeContainerId);
+  if (iframeContainer !== null) {
+    iframeContainer.innerHTML = "";
+    iframeContainer.style.display = "none";
+  } else {
+    iframeContainer = document.createElement("div");
+    iframeContainer.id = iframeContainerId;
+    iframeContainer.style.display = "none";
+    document.body.appendChild(iframeContainer);
+  }
+
+  const signer = new AlchemySigner({
+    client: {
+      ...client,
+      iframeConfig: {
+        ...client.iframeConfig,
+        iframeContainerId,
+      },
+    },
+    sessionConfig,
+  });
+
+  const search = new URLSearchParams(window.location.search);
+  if (search.has("bundle")) {
+    signer.authenticate({ type: "email", bundle: search.get("bundle")! });
+  }
+
+  return signer;
+};
+
+const getSignerStatus = (
+  alchemySignerStatus: AlchemySignerStatus
+): SignerStatus => ({
+  status: alchemySignerStatus,
+  isInitializing: alchemySignerStatus === AlchemySignerStatus.INITIALIZING,
+  isAuthenticating:
+    alchemySignerStatus === AlchemySignerStatus.AUTHENTICATING ||
+    alchemySignerStatus === AlchemySignerStatus.AWAITING_EMAIL_AUTH,
+  isConnected: alchemySignerStatus === AlchemySignerStatus.CONNECTED,
+  isDisconnected: alchemySignerStatus === AlchemySignerStatus.DISCONNECTED,
+});
+
+// This is done this way to avoid issues with React requiring static state
+const staticState: AccountState<SupportedAccountTypes> = {
+  status: "DISCONNECTED",
+  account: undefined,
+};
+
+export const defaultAccountState = <
+  T extends SupportedAccountTypes
+>(): AccountState<T> => staticState;
```

Comment:
does this need a generic?

Reply:
yea because we don't have a default account type in the code base. if we want to default to MOMA or LA then we can update this to work on that generic